gamification-github-services/src/main/java/org/exoplatform/gamification/github/dao/WebHookDAO.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ * Copyright (C) 2020 - 2023 Meeds Association contact@meeds.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.exoplatform.gamification.github.dao;
+
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.gamification.github.entity.WebhookEntity;
+
+import javax.persistence.NoResultException;
+import javax.persistence.TypedQuery;
+import java.util.Date;
+import java.util.List;
+
+public class WebHookDAO extends GenericDAOJPAImpl<WebhookEntity, Long> {
+
+  public static final String ORGANIZATION_ID = "organizationId";
+
+  public WebhookEntity getWebhookByOrganizationId(long organizationId) {
+    TypedQuery<WebhookEntity> query = getEntityManager().createNamedQuery("GitHubWebhooks.getWebhookByOrganizationId",
+                                                                          WebhookEntity.class);
+    query.setParameter(ORGANIZATION_ID, organizationId);
+    query.setMaxResults(1);
+    try {
+      return query.getSingleResult();
+    } catch (NoResultException e) {
+      return null;
+    }
+  }
+
+  @Override
+  @ExoTransactional
+  public WebhookEntity update(WebhookEntity entity) {
+    entity.setUpdatedDate(new Date());
```

Comment:
It's better to set this logic in Storage layer. When moving to Spring Container by example, the DAO layer remains unalterable.

```suggestion
```